### PR TITLE
Itineraries

### PR DESCRIPTION
**Pull request by hannahkm**
[bugfixes] : misc bug fixes to improve ux 

Summary: users were previously able to edit/delete day markers, which would then crash the app. Editing destinations/itineraries would also not show all previous information in the form fields. Also, when registering, users would not be given an empty favorites list, which would then cause the app to crash when they try to access their favorites page. This PR fixes these problems.

Changes in this PR:
- Destinations onclick will first check if the item being selected is a day marker, and only fire the onclick method if the item is a destination created by the user.
- Similar change to onLongClick; the method is only fired when the user is selecting a user-made destination, so they cannot delete the day markers
- When registering for an account, the user is also initialized with an empty list for their favorites and for their itineraries, so the app will not crash when trying to access a null variable

### CODE REVIEW

File: app/src/main/java/com/example/fbufinalapp/CommonFunctions.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.example.fbufinalapp;
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/CommonFunctions.java:1]: Overall, great job centralizing some functions. That's a great coding habit that you should continue doing :)

File: app/src/main/java/com/example/fbufinalapp/CommonFunctions.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+public class CommonFunctions {
+    static Place place;
+
+    public static Place getPlace(String placeId, Context context) {
+        PlacesClient placesClient = Places.createClient(context);
+
+        // Specify the fields to return.
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/CommonFunctions.java:21]: comment unneeded

File: app/src/main/java/com/example/fbufinalapp/CommonFunctions.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        List<Place.Field> placeFields = Arrays.asList(Place.Field.ID,
+                Place.Field.NAME, Place.Field.ADDRESS, Place.Field.RATING, Place.Field.PRICE_LEVEL,
+                Place.Field.WEBSITE_URI, Place.Field.PHONE_NUMBER);
+
+        // Construct a request object, passing the place ID and fields array.
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/CommonFunctions.java:26]: comment unneeded

File: app/src/main/java/com/example/fbufinalapp/DashboardFragment.java [hunk 4]
```diff
@@ -149,6 +150,8 @@ public void fetchTimelineAsync(int page) {
     private void queryPosts() {
         ParseQuery<Itinerary> query = ParseQuery.getQuery(Itinerary.class);
         query.include(Itinerary.KEY_USER);
+        query.whereEqualTo("authors", ParseUser.getCurrentUser());
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/DashboardFragment.java:153]: Is there a place where the keys for a user is kept? 
If so, then it is best to, similar to reference that code to have a centralized place for all the ParseUser fields. 
If not,  it's a good idea to have a class enum for ParseUsers that has all the fields. Something like this  :

```
class User {
ID = "id"
NAME = "name"
AUTHORS = "authors" 
}
```

This document would explain how best to create an enum type to do something along those lines in Java. 
https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

File: app/src/main/java/com/example/fbufinalapp/EditDestinationActivity.java [hunk 5]
```diff
@@ -93,6 +93,15 @@ protected void onCreate(Bundle savedInstanceState) {
                 query.getInBackground(getIntent().getStringExtra("destinationId"), (object, e) -> {
                     if (e == null) {
                         editingDestination = object;
+                        Date date = editingDestination.getDate();
+
+                        SimpleDateFormat dateFormatter = new SimpleDateFormat("MMM dd, yyyy");
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/EditDestinationActivity.java:98]: is `MMM` a typo?

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java [hunk 6]
```diff
@@ -81,4 +87,28 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
     public void setActionBarTitle(String title) {
         getSupportActionBar().setTitle(title);
     }
+
+    public static Place getPlace(String placeId, Context context) {
+        PlacesClient placesClient = Places.createClient(context);
+
+        // Specify the fields to return.
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/MainActivity.java:94]: unneeded comment

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java [hunk 7]
```diff
@@ -81,4 +87,28 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
+        List<Place.Field> placeFields = Arrays.asList(Place.Field.ID,
+                Place.Field.NAME, Place.Field.ADDRESS, Place.Field.RATING, Place.Field.PRICE_LEVEL,
+                Place.Field.WEBSITE_URI, Place.Field.PHONE_NUMBER);
+
+        // Construct a request object, passing the place ID and fields array.
+        FetchPlaceRequest request = FetchPlaceRequest.newInstance(placeId, placeFields);
+
+        placesClient.fetchPlace(request).addOnSuccessListener((response) -> {
+            place = response.getPlace();
+            Log.i("CommonFunctions", "inside: " + String.valueOf(place));
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/MainActivity.java:104]: unneeded comment

File: app/src/main/java/com/example/fbufinalapp/DashboardFragment.java (hunk 4 above)
Comment (hannahkm) [reply app/src/main/java/com/example/fbufinalapp/DashboardFragment.java:153]: yes! I can add these to a central file :)

File: app/src/main/java/com/example/fbufinalapp/EditDestinationActivity.java (hunk 5 above)
Comment (hannahkm) [reply app/src/main/java/com/example/fbufinalapp/EditDestinationActivity.java:98]: Yes, thank you for catching that! I've fixed it now :)

**Pull request merged**